### PR TITLE
fmu-v6xrt lpuart singlewire for RC telemetry

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Updates IMXRT driver to support RC telemetry.

Also changes rc_input singlewire mode to pushpull for better signalling.

~~Pending on PX4/NuttX#307~~

### CODE REVIEW

**Comment by davids5** [comment]
Pending on https://github.com/PX4/NuttX/pull/307 - is merged - please repoint this PR

File: boards/px4/fmu-v6xrt/src/board_config.h [hunk 1]
```diff
@@ -451,10 +451,12 @@
...
 
-#define RC_SERIAL_PORT                  "/dev/ttyS4"
-#define RC_SERIAL_SINGLEWIRE            1 // Suport Single wire wiring
-#define RC_SERIAL_SWAP_RXTX             1 // Set Swap (but not supported in HW) to use Single wire
-#define RC_SERIAL_SWAP_USING_SINGLEWIRE 1 // Set to use Single wire swap as HW does not support swap
+#define RC_SERIAL_PORT                     "/dev/ttyS4"
+#define RC_SERIAL_SINGLEWIRE               1 // Suport Single wire wiring
+#define RC_SERIAL_SWAP_RXTX                1 // Set Swap (but not supported in HW) to use Single wire
+#define RC_SERIAL_SWAP_USING_SINGLEWIRE    1 // Set to use Single wire swap as HW does not support swap
+#define RC_SERIAL_SINGLEWIRE_MANUAL_DUPLEX 1 // Set to use ioctl's to manually swap duplex
```
Comment (davids5) [inline boards/px4/fmu-v6xrt/src/board_config.h:458]: Needed?

**Review by davids5 — CHANGES_REQUESTED** [review]

File: boards/px4/fmu-v6xrt/src/board_config.h (hunk 1 above)
Comment (PetervdPerk-NXP) [reply boards/px4/fmu-v6xrt/src/board_config.h:458]: Indentation change, unless you want to keep it as is.

Comment (davids5) [reply boards/px4/fmu-v6xrt/src/board_config.h:458]: But is it used anywhere?

Comment (PetervdPerk-NXP) [reply boards/px4/fmu-v6xrt/src/board_config.h:458]: https://github.com/PX4/PX4-Autopilot/commit/93a256741ff672a8e863d9fc62c523c0b989b2c2

Comment (davids5) [reply boards/px4/fmu-v6xrt/src/board_config.h:458]: RC_SERIAL_SINGLEWIRE_MANUAL_DUPLEX  ?

Comment (PetervdPerk-NXP) [reply boards/px4/fmu-v6xrt/src/board_config.h:458]: Good point that's unused and a leftover from the 1st version

**Comment by dk7xe** [comment]
nsh: sysinit: fopen failed: No such file or directory

NuttShell (NSH) NuttX-11.0.0
nsh> ver all
HW arch: PX4_FMU_V6XRT
HW type: V6XRT
HW FMUM ID: 0x000
HW BASE ID: 0x003
PX4 git-hash: bbaed84123bd51db9ef95fd34a66dd91513b467c
PX4 version: 1.15.0 80 (17760384)
PX4 git-branch: pr-lpuart-singlewire-v6xrt
OS: NuttX
OS version: Release 11.0.0 (184549631)
OS git-hash: f8b59abbf98e19a184114c3f54462810ed5990ab
Build datetime: May 25 2024 20:58:17
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 00090000000000000000829297441d1a180e
MCU: i.MX RT1170 rB0, rev. 0
nsh> 

nsh> rc_input status
INFO  [rc_input] Max update rate: 250 Hz
INFO  [rc_input] UART device: /dev/ttyS4
INFO  [rc_input] UART RX bytes: 270691
INFO  [rc_input] RC state: searching for signal: GHST
rc_input: cycle time: 30492 events, 621143us elapsed, 20.37us avg, min 6us max 870us 30.530us rms
rc_input: publish interval: 18790 events, 4096.65us avg, min 1785us max 15986us 680.536us rms
nsh> 

@PetervdPerk-NXP but no telemetry on my Tandem X20S other than from the Ghost Module itself.
On my racers with FETtec FC i do get telemetry

**Comment by dk7xe** [comment]
After switching the GHST protocol from Race250 to Normal telemetry works

**Comment by dk7xe** [comment]
After switching the GHST protocol from Race250 to Normal telemetry works


Switched back to Race250 protocol.
Now telemetry works as well after rebinding the Ghost Atto Duo RX

**Comment by dk7xe** [comment]
To me it looks like this is good to go

**Comment by PetervdPerk-NXP** [comment]
Pending on testing on Pixhawk 4 - FMUv5 due to `SER_SINGLEWIRE_PUSHPULL` change.
@dk7xe you happen to have a Pixhawk 4?

**Comment by PetervdPerk-NXP** [comment]
Dropped `SER_SINGLEWIRE_PUSHPULL` change it's working fine without it.
Made a seperate PR for that see #23216

@davids5 can we merge this?

**Review by davids5 — APPROVED** [review]

**Review by davids5 — APPROVED** [review]

**Pull request merged**